scripts/release/build/image_build_process.py

```diff
@@ -102,11 +116,33 @@ def execute_docker_build(
         # Convert build args to the format expected by python_on_whales
         build_args = {k: str(v) for k, v in args.items()}
 
+        registry_name = tag.split(":")[0] if ":" in tag else tag
+        # e.g., "268558157000.dkr.ecr.us-east-1.amazonaws.com/dev/mongodb-kubernetes" -> "mongodb-kubernetes"
+        cache_image_name = registry_name.split("/")[-1]
+        # TODO CLOUDP-335471: use env variables to configure AWS region and account ID
+        cache_repo_name = f"dev/cache/{cache_image_name}"
+        cache_registry = f"268558157000.dkr.ecr.us-east-1.amazonaws.com/{cache_repo_name}"
+        cache_from = f"type=registry,ref={f"{cache_registry}:cache"}"
+
+        cache_to = {
+            "type": "registry",
+            "ref": f"{cache_registry}:cache",
+            "mode": "max",
+            "oci-mediatypes": "true",
+            "image-manifest": "true"
```

Comment:
"""
By default, the OCI media type generates an image index for the cache image. Some OCI registries, such as Amazon ECR, don't support the image index media type: application/vnd.oci.image.index.v1+json. If you export cache images to ECR, or any other registry that doesn't support image indices, set the image-manifest parameter to true to generate a single image manifest instead of an image index for the cache image:
"""